rust/lance/src/dataset/transaction.rs

```diff
@@ -1626,7 +1631,21 @@ impl Transaction {
                 if let Some(next_row_id) = &mut next_row_id {
                     Self::assign_row_ids(next_row_id, new_fragments.as_mut_slice())?;
                 }
+                // Identify fragments that were updated or newly created in this update
+                let mut target_ids: HashSet<u64> = HashSet::new();
+                target_ids.extend(updated_fragments.iter().map(|f| f.id));
+                target_ids.extend(new_fragments.iter().map(|f| f.id));
+
                 final_fragments.extend(new_fragments);
+                if config.use_stable_row_ids {
+                    let current_version = self.read_version + 1;
+                    // Override version metadata for updated/new fragments to mark this change
```

Comment:
One important thing I just realize I forgot to mention, I think for any new fragments, the `RowLatestUpdateVersionMeta` should remain null. This saves computational effort and also makes it more backwards compatible. Essentially, if the fragment does not contain this information, it is considered that all rows are added in the latest version.

Reply:
Done.